scripts/jarvis.js

```diff
@@ -0,0 +1,165 @@
+const fs = require('fs');
+const { spawn } = require('child_process');
+
+const nodeModulesExist = fs.existsSync('node_modules');
+const envInitialized = fs.existsSync('apps/api/src/.env');
+
+if (!nodeModulesExist || !envInitialized) {
```

Comment:
This is so cool idea to have a tool that is helping to manage the project :) I really like it ;) 

However, the current approach has one disadvantage you need to run it two times if you are just bootstrapping the project for the first time... and it's hard to notice what it says at the beginning... 

if I can make a suggestion here, I would change the code a little to smth like this:
- create the `async function main` where all code will be running
- create the `informAboutInitialSetup` function that will return a Promise which will use the NodeJS API (because node_modules are not installed yet) to inform and ask the question `Looks like its the first time... Would you like to continue? Yes/No`
- wrap the `spawn` code into the `setupProject` function that returns the `Promise` or rejects on `exit`
- wrap the `jarvis` questions code to another async function like `askQuestions`
- from the `main` call the `informAboutInitialSetup, setupProject, askQuestions`

Here is the suggested code, it might require some more improvements:
```
const informAboutInitialSetup = () =>
  new Promise((resolve, reject) => {
    const rl = readline.createInterface({
      input: process.stdin,
      output: process.stdout,
    });

    rl.question(
      'Looks like its the first time running this project on your machine. We will start by installing pnpm depependencies. Do you want to continue? Yes/No\n',
      function (answer) {
        if (answer.toLowerCase() === 'yes' || answer.toLowerCase() === 'y') {
          rl.close();
          resolve();
          return;
        }

        reject();
      }
    );
  });

const setupProject = () =>
  new Promise((resolve, reject) => {
    const isWindows = process.platform === 'win32';
    const cmd = isWindows ? 'cmd' : 'npm';
    const args = isWindows ? ['/c', 'npm run setup:project'] : ['run setup:project'];

    const command = spawn(cmd, args, {
      shell: true,
      stdio: 'inherit',
    });

    function onExit() {
      command.kill('SIGINT');
    }

    process.on('SIGTERM', onExit);
    process.on('SIGINT', onExit);

    command.on('exit', (exitCode) => {
      if (parseInt(exitCode) !== 0) {
        reject(new Error(exitCode));
      }

      console.log('FINISHED!');
      resolve();
    });
  });

async function askQuestions() {
// questions code here
}

async function main() {
  if (!nodeModulesExist || !envInitialized) {
    await informAboutInitialSetup();

    await setupProject();
  }

  await askQuestions();
}

main().catch(() => {
  process.exit(1);
});
```

you can give better names ;)

Reply:
Really loved your approach, and at your refactor after the installation the program is not exit (it continue to the askQuestions part)